modules/modal/react/lib/Modal.tsx

```diff
@@ -69,125 +69,124 @@ const transformOrigin = {
-          // We simply want to transfer focus inside
-          firstFocusable.removeEventListener('blur', changeTabIndex);
-          // We must wait one frame to ensure tabbable checks are satisfied
-          // by all focus libraries...
-          requestAnimationFrame(() => {
-            firstFocusable.removeAttribute('tabIndex');
-          });
-        };
-        firstFocusable.addEventListener('blur', changeTabIndex);
-      }
-      return firstFocusable;
-    } else {
-      throw new Error(
-        'No focusable element was found. Please ensure modal has at least one focusable element'
-      );
+function getFirstElementToFocus(modalEl: HTMLElement): HTMLElement {
+  const firstFocusable = modalEl.querySelector<HTMLElement>(
+    `[data-close=close],[id="${modalEl.getAttribute('aria-labelledby')}"]`
+  );
+  if (firstFocusable) {
+    if (firstFocusable.tagName === 'H3') {
+      // If there is no close icon, we need to transfer focus to the header.
+      // Setting tabIndex allows the header to be focusable.
+      // We do the header instead of the next focusable element to prevent useful context from being skipped
+      firstFocusable.tabIndex = 0;
+      firstFocusable.style.outline = 'none';
+
+      const changeTabIndex = () => {
+        // We no longer need to focus on the header after it loses focus
+        // We simply want to transfer focus inside
+        firstFocusable.removeEventListener('blur', changeTabIndex);
+        // We must wait one frame to ensure tabbable checks are satisfied
+        // by all focus libraries...
+        requestAnimationFrame(() => {
+          firstFocusable.removeAttribute('tabIndex');
+        });
+      };
+      firstFocusable.addEventListener('blur', changeTabIndex);
     }
+    return firstFocusable;
+  } else {
+    throw new Error(
+      'No focusable element was found. Please ensure modal has at least one focusable element'
+    );
   }
 }
 
-export default class Modal extends React.Component<ModalProps> {
-  private modalRef = React.createRef<HTMLDivElement>();
-
-  static Padding = PopupPadding;
-  static Width = ModalWidth;
-  static defaultProps = {
-    open: false,
-    padding: Modal.Padding.l,
-    width: Modal.Width.s,
-    closeOnEscape: true,
-  };
+const useKeyDownListener = (handleKeydown: EventListenerOrEventListenerObject) => {
+  React.useEffect(() => {
+    document.addEventListener('keydown', handleKeydown);
+    return () => {
+      document.removeEventListener('keydown', handleKeydown);
+    };
+  }, []);
+};
+
+const useInitialFocus = (
+  modalRef: React.RefObject<HTMLElement>,
+  firstFocusableRef: React.RefObject<HTMLElement> | undefined
+) => {
+  React.useEffect(() => {
+    if (modalRef.current) {
+      const elem =
+        (firstFocusableRef && firstFocusableRef.current) ||
+        getFirstElementToFocus(modalRef.current);
+      elem.focus();
+    }
+  }, [modalRef]);
+};
+
+const Modal = ({
+  open,
```

Comment:
This PR removes default props, but doesn't set them back up again here. Adding `open = false` would replace existing functionality of the default props.